homeassistant/components/eq3btsmart/climate.py

```diff
@@ -106,52 +343,106 @@ def temperature_unit(self):
 
     @property
     def precision(self):
-        """Return eq3bt's precision 0.5."""
-        return PRECISION_HALVES
+        """Return the precision of the system."""
+        if self._temp_precision is not None:
+            return self._temp_precision
+
+        return super().precision
 
     @property
     def current_temperature(self):
-        """Can not report temperature, so return target_temperature."""
+        """Return the sensor temperature if defined, otherwise return target_temperature."""
+        if self._sensor_entity_id:
+            return self._cur_temp
+
         return self.target_temperature
 
     @property
     def target_temperature(self):
         """Return the temperature we try to reach."""
+        if self._sensor_control:
+            return self._target_temp
+
         return self._thermostat.target_temperature
 
-    def set_temperature(self, **kwargs):
+    async def async_set_temperature(self, **kwargs):
         """Set new target temperature."""
         temperature = kwargs.get(ATTR_TEMPERATURE)
         if temperature is None:
             return
+
+        if self._sensor_control:
+            self._target_temp = temperature
+            await self._async_control_heating()
+            self.async_write_ha_state()
+            return
+
         self._thermostat.target_temperature = temperature
 
     @property
     def hvac_mode(self):
         """Return the current operation mode."""
+        if self._sensor_control:
+            return self._hvac_mode
+
         if self._thermostat.mode < 0:
             return HVAC_MODE_OFF
         return EQ_TO_HA_HVAC[self._thermostat.mode]
 
+    @property
+    def hvac_action(self):
+        """Return the current running hvac operation if supported.
+
+        Need to be one of CURRENT_HVAC_*.
+        """
+        if not self._sensor_control:
+            return None
+
+        if self.hvac_mode == HVAC_MODE_OFF:
+            return CURRENT_HVAC_OFF
+
+        return EQ_TO_HA_HVAC_ACTION.get(self._thermostat.mode, "-")
```

Comment:
The docstring indicates that the return value needs to be `CURRENT_HVAC_*` but this falls back to `-` on unknown modes.

Reply:
I wanted to show some invalid state in case the thermostat is not in the expected state, and could not find a suitable value in CURRENT_HVAC_*.
So I suspect this is not ok. What would be the correct approach ?

Reply:
Sorry, I don't know about that, I did just briefly checked parts of the diff and saw that there was a discrepancy between the docstring and the code :-(